### PR TITLE
Added a bunch of tests for courses endpoints

### PR DESCRIPTION
**Pull request by DonIsaac**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,7 +29,11 @@ group :test do
   gem 'rack-test', require: 'rack/test'
   gem 'simplecov', require: false
   gem 'codecov', :require => false, :group => :test
-  gem 'json-schema'
```
Comment (McIntireEvan) [inline Gemfile:32]: is removing this intentional?

File: tests/v1/courses/courses_spec.rb [hunk 2]
```diff
@@ -8,15 +11,132 @@ def build_url1(u)
...
     describe 'GET /courses' do
       before { get(build_url1('?')) }
+
+      let(:res) { JSON.parse(last_response.body) }
+
       it_has_behavior 'good status', (build_url1 '?')
+
       it 'returns a list of courses' do
-        res = JSON.parse(last_response.body)
+        # res = JSON.parse(last_response.body)
```
Comment (McIntireEvan) [inline tests/v1/courses/courses_spec.rb:20]: either uncomment or delete

File: tests/v0/map_spec.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require_relative '../spec_helper'
 
-describe 'Map Endpoints v1' do
-  url = 'v1/map'
+describe 'Map Endpoints v0' do
+  url = 'v0/map'
```
Comment (McIntireEvan) [inline tests/v0/map_spec.rb:4]: good fix - I see for map spec v1, you add a leading `/`, can you do that for this/others to be consistent?

**Review by McIntireEvan — APPROVED** [review]
I left a few comments, but overall looks good. Approving to unblock you, but would like to see those addressed before merging.

Thanks for taking testing so seriously!

File: Gemfile (hunk 1 above)
Comment (DonIsaac) [reply Gemfile:32]: yes

File: tests/v0/map_spec.rb (hunk 3 above)
Comment (DonIsaac) [reply tests/v0/map_spec.rb:4]: yes

**Review by McIntireEvan — APPROVED** [review]

**Pull request merged**